cmd/syncthing/gui.go

```diff
@@ -1024,6 +1027,87 @@ func (s *apiService) getSystemLogTxt(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+type fileEntry struct {
+	name string
+	data []byte
+}
+
+func (s *apiService) getSupportBundle(w http.ResponseWriter, r *http.Request) {
+	var arrayFile []fileEntry
+
+	// Redacted configuration as a JSON
+	jsonConfig, _ := json.MarshalIndent(getRedactedConfig(s), "", "  ")
+	arrayFile = append(arrayFile, fileEntry{name: "config.json", data: jsonConfig})
+
+	// Log as a text
+	var buflog bytes.Buffer
+	for _, line := range s.systemLog.Since(time.Time{}) {
+		fmt.Fprintf(&buflog, "%s: %s\n", line.When.Format(time.RFC3339), line.Message)
+	}
+	arrayFile = append(arrayFile, fileEntry{name: "log.txt", data: buflog.Bytes()})
+
+	// Errors as a JSON
+	if errs := s.guiErrors.Since(time.Time{}); len(errs) > 0 {
+		jsonError, _ := json.MarshalIndent(errs, "", "  ")
+		arrayFile = append(arrayFile, fileEntry{name: "errors.json", data: jsonError})
+	}
+
+	// Errors as a JSON
+	if panicFiles, err := filepath.Glob(filepath.Join(baseDirs["config"], "panic*")); err == nil {
+		for _, f := range panicFiles {
+			if panicF, err := ioutil.ReadFile(f); err == nil {
+				arrayFile = append(arrayFile, fileEntry{name: filepath.Base(f), data: panicF})
+			}
+		}
+	}
+
+	// Version and platform information as a JSON
+	versionPlatform, _ := json.MarshalIndent(map[string]string{
+		"version":     Version,
+		"codename":    Codename,
+		"longVersion": LongVersion,
+		"os":          runtime.GOOS,
+		"arch":        runtime.GOARCH,
+	}, "", "  ")
+	arrayFile = append(arrayFile, fileEntry{name: "versionPlatform.json", data: versionPlatform})
+
+	// Report Data as a JSON
+	usageReportingData, _ := json.MarshalIndent(reportData(s.cfg, s.model, s.connectionsService, usageReportVersion, true), "", "  ")
+	arrayFile = append(arrayFile, fileEntry{name: "usageReportingData.json", data: usageReportingData})
+
+	// Heap and CPU Proofs as a pprof extension
+	var heapBuffer, cpuBuffer bytes.Buffer
+	filename := fmt.Sprintf("syncthing-heap-%s-%s-%s-%s.pprof", runtime.GOOS, runtime.GOARCH, Version, time.Now().Format("150405")) // hhmmss
+	runtime.GC()
+	pprof.WriteHeapProfile(&heapBuffer)
+	arrayFile = append(arrayFile, fileEntry{name: filename, data: heapBuffer.Bytes()})
+
+	const duration = 4 * time.Second
+	filename = fmt.Sprintf("syncthing-cpu-%s-%s-%s-%s.pprof", runtime.GOOS, runtime.GOARCH, Version, time.Now().Format("150405")) // hhmmss
+	pprof.StartCPUProfile(&cpuBuffer)
+	time.Sleep(duration)
+	pprof.StopCPUProfile()
+	arrayFile = append(arrayFile, fileEntry{name: filename, data: cpuBuffer.Bytes()})
+
+	// Add files to buffer zip
+	var newfileBuffer bytes.Buffer
+	zipBuffer(&newfileBuffer, arrayFile)
+
+	output := fmt.Sprintf("support-bundle-%s.zip", time.Now().Format("2006-01-02T15.04.05"))
```

Comment:
This is not really output, this is just a file name.

Reply:
yeap, I use filename for heapprof and cpuprof but I agree that output is not appropriate, going to change...